state/cloudimagemetadata/interface.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package cloudimagemetadata
+
+// Metadata describes a cloud image metadata.
+type Metadata struct {
+	Storage     string
+	VirtType    string
+	Arch        string
+	Series      string
+	RegionAlias string
+	RegionName  string
+	Endpoint    string
+	Stream      string
+}
+
+// Storage provides methods for storing and retrieving cloud image metadata.
+type Storage interface {
```

Comment:
Renamed Metadata(series, arch, stream) to FindMetadata(series, arch, stream) for clarity.